zerver/models.py

```diff
@@ -1486,6 +1486,17 @@ class UserBaseSettings(models.Model):
     display_emoji_reaction_users: bool = models.BooleanField(default=True)
     twenty_four_hour_time: bool = models.BooleanField(default=False)
     starred_message_counts: bool = models.BooleanField(default=True)
+    EMOJI_ANIMATION_CONFIG_NEVER = 1
+    EMOJI_ANIMATION_CONFIG_ON_HOVER = 2
+    EMOJI_ANIMATION_CONFIG_ALWAYS = 3
+    EMOJI_ANIMATION_CONFIG_CHOICES = [
+        EMOJI_ANIMATION_CONFIG_NEVER,
+        EMOJI_ANIMATION_CONFIG_ON_HOVER,
+        EMOJI_ANIMATION_CONFIG_ALWAYS,
+    ]
+    emoji_animation_config: int = models.PositiveSmallIntegerField(
+        default=EMOJI_ANIMATION_CONFIG_ON_HOVER
+    )
```

Comment:
If you can extract just the backend changes here as one of the prep commits, I can probably merge that (I think we're reasonably happy with the API design) and stop having to deal with merge conflicts while we rebase the rest of the feature.